### ISSUE TITLE
.setChromeOptions() not respected with chromedriver@75

### ISSUE THREAD

**Issue by rensbaardman** [SeleniumHQ/selenium#7289]
## 🐛 Bug Report

This is about `selenium-webdriver` for Node.

In some cases, when trying to create a new driver for Chrome, options set with `.setChromeOptions()` are not respected. If you have a Node script that creates the driver, it depends on how the script is called: the options are respected if you call `node ...` directly, but it does not work if you call the same script via `npm run ...` with a command aliased to that same script.

## To Reproduce
Take the following script:

```javascript
// test.js

const {Builder} = require('selenium-webdriver');
const chrome = require('selenium-webdriver/chrome');

// 'disable-infobars' should disable the "Chrome is being controlled
// by automated test software"-warning
let chrome_options = new chrome.Options().addArguments("disable-infobars")

const driver = new Builder()
	.setChromeOptions(chrome_options)
	.forBrowser('chrome')
	.build();
```

If you run it with
```
node test.js
```
a Chrome browser should come up, with **no** warning about "Chrome is being controlled by automated test software.".

Now run the same script via
```
npm test
```
where we aliased `test` to `node test.js`:
```javascript
// package.json

...
"scripts": {
	"test": "node test.js"
}
```
Again, a Chrome browser should pop up, but now (in my case), it **does** have the warning ""Chrome is being controlled by automated test software.".

## Expected behavior

In both cases, the `.setChromeOptions()` should be respected, and no warning should be shown.

I suspect the issue is not with `npm` itself, but that it does set certain environment variables that influence how the driver is built. E.g. if you dump `process.env` in the second case, `npm` has set tens of `npm_config` environment variables. Something like that might be the cause. In that case, I expect `selenium-webdriver` to handle that more robustly.

Note: `npm` runs scripts with `sh`, and although I use `zsh`, if you use `sh` specifically:
```
sh -c 'node test.js`
```
it still works, so I don't think that's the problem.

No problems arise when doing the same in Firefox (i.e. with `.setFirefoxOptions`)

If there are any other ways I can test the behaviour of the webdriver, I would like to know, as it was very hard for me to properly debug the issue and check how and why webdriver does certain things. I did manage to [get the output of the `webdriver.http` logs,](https://seleniumhq.github.io/selenium/docs/api/javascript/module/selenium-webdriver/lib/logging.html) and although it differed between the two cases, I couldn't tell what caused it and whether the difference was problematic.

## Test script or set of commands reproducing this issue

For a full setup demonstrating the bug, see my repo [rensbaardman/npm-selenium-webdriver-bug](https://github.com/rensbaardman/npm-selenium-webdriver-bug).

## Environment

OS: OS X 10.11.6
Browser: Chrome
Browser version: 75.0.3770.80
Browser Driver version: chromedriver@75.0.0
selenium-webdriver version: selenium-webdriver@4.0.0-alpha.1
Node version: v10.16.0
npm version: 6.9.0

**Comment by rensbaardman** [comment]
I did some extra tests, and found out that my suspicion was correct: it turns out that `npm run` modifies the `PATH` environment variable, so that `chromedriver` was now executed from `node_modules/.bin/chromedriver` which aliases to `node_modules/chromedriver/bin/chromedriver`. This uses the latest version of `chromedriver`, `75.0.3770.8 (681f24ea911fe754973dda2fdc6d2a2e159dd300-refs/branch-heads/3770@{#40})`.

When running directly with Node, `chromedriver` was executed from `/usr/local/bin` which contained an **old** version of `chromedriver`: `2.46.628411 (3324f4c8be9ff2f70a05a30ebc72ffb013e1a71e)`. The question is then: why did the old version of `chromedriver` work, and then new one didn't?

**Comment by rensbaardman** [comment]
Alright, I found the culprit! Since version 75, chromedriver [is apparently defaulting to 'W3C mode',](https://groups.google.com/forum/#!topic/chromedriver-users/ZnGlzYfsgt4) which requires chrome options to be communicated (in the wire protocol) with `'goog:chromeOptions'` instead of the previous `'chromeOptions'`. This was changed in `selenium-webdriver` with https://github.com/SeleniumHQ/selenium/commit/4c8646cde1dcebbf2665b676aa5ce010c6598dce (already a year ago), but apparently the version of `selenium-webdriver` published to `npm` hasn't been updated for over a year!

See https://github.com/SeleniumHQ/selenium/issues/6394 for more on a new version on `npm`.
